chains/evm/contracts/ocr/SignatureQuorumVerifier.sol

```diff
@@ -2,150 +2,169 @@
-  event ConfigSet(bytes32 configDigest, address[] signers, uint8 F);
+  /// @param F maximum number of faulty/dishonest singers the protocol can tolerate while still working correctly.
+  event ConfigSet(bytes32 indexed configDigest, address[] signers, uint8 F);
+  event ConfigRevoked(bytes32 indexed configDigest);
 
   error InvalidConfig();
-  error ConfigDigestMismatch(bytes32 expected, bytes32 actual);
+  error InvalidConfigDigest(bytes32 configDigest);
+  error ConfigDigestAlreadyExists(bytes32 configDigest);
   error ForkedChain(uint256 expected, uint256 actual);
   error WrongNumberOfSignatures();
   error SignaturesOutOfRegistration();
   error UnauthorizedSigner();
   error NonUniqueSignatures();
   error OracleCannotBeZeroAddress();
 
-  struct SignatureConfigConfig {
-    bytes32 configDigest;
-    uint8 F; //  maximum number of faulty/dishonest oracles the system can tolerate.
-    uint8 n; //  number of configured signers.
-  }
-
   /// @notice Args to update an OCR Config.
   struct SignatureConfigArgs {
     bytes32 configDigest; // The new config digest.
     uint8 F; // Maximum number of faulty/dishonest oracles.
     address[] signers; // signing address of each oracle.
   }
 
+  struct SignatureConfigConfig {
+    uint8 F; //  Maximum number of faulty/dishonest oracles the system can tolerate.
+    EnumerableSet.AddressSet signers;
+  }
+
   struct SignatureProof {
     bytes32[] rs; // R components of the signatures.
     bytes32[] ss; // S components of the signatures.
   }
 
-  SignatureConfigConfig internal s_ocrConfig;
-
-  EnumerableSet.AddressSet internal s_signers;
+  mapping(bytes32 configDigest => SignatureConfigConfig config) private s_signatureConfig;
+  EnumerableSet.Bytes32Set private s_activeConfigDigests;
 
   uint256 internal immutable i_chainID;
 
   constructor() {
     i_chainID = block.chainid;
   }
 
-  // TODO allow older digests.
-  function _validateConfigDigest(
-    bytes32 configDigest
-  ) internal view {
-    if (s_ocrConfig.configDigest != configDigest) {
-      revert ConfigDigestMismatch(s_ocrConfig.configDigest, configDigest);
+  /// @notice Validates the signatures of a given report hash.
+  /// @dev The configDigest should be included in the report hash!
+  /// @param reportHash The hash of the report to validate signatures for.
+  /// @param configDigest The config digest determines the signature set to use for validation.
+  /// @param signatureProof The signatures to validate, encoded as a SignatureProof struct.
+  function _validateSignatures(bytes32 reportHash, bytes32 configDigest, bytes memory signatureProof) internal view {
+    // We allow proving of older messages that might have been signed by an older set of signers. This means we need to
+    // get the set of signers for the given configDigest.
+    SignatureConfigConfig storage config = s_signatureConfig[configDigest];
```

Comment:
Ah no config digest means doing this old signer check is kinda hard. I guess we haven't really landed on the end state of this part of the commit verifier?

In the DD we're basically saying we have to resign in certain cases and we just don't support old signers. I'm not sure the full implications of that, in the steady state, it seems like it should be OK if we have more lenient signature checks as described in [the DD](https://docs.google.com/document/d/1hD0s6YnpWr17sqpxZZmgLnRwt7XdL6Q5Z9I5iCrgJB8/edit?tab=t.0#heading=h.axfiogfv4b3u)